### DISCUSSION TITLE
Mass-locking of all objects pre- and post-simulation

### DISCUSSION THREAD

**Discussion by SSNTails** [jrouwe/JoltPhysics#993] [Q&A] [answered]
Is there a performance penalty to updating each object's position, velocity, etc. on an individual lock basis as so, or is there some way to 'mass lock' the entire simulation both before (to update the positions and/or velocity) and after the simulation (to obtain the results)? I see some of the samples grab this information on a per-object basis, but is this best practice, or just done for the sake of a small sample?

`

for each object in listOfObjects....
JPH::BodyInterface &bi = mPhysicsSystem->GetBodyInterface();
bi.SetRotation(obj->collisionObjectID, mat.GetQuaternion().Normalized(), JPH::EActivation::Activate); // Requests a lock each time this is called.

`

**Comment by jrouwe** [comment, marked as answer]
Read my reply here and let me know if there are still questions:

https://github.com/jrouwe/JoltPhysics/issues/989

**Comment by jrouwe** [comment]
B.t.w. mass changing the rotations of all objects is probably not what you want because it means that the physics system doesn't know about the velocity of rotation, so you won't get correct physical interaction. Take a look at `Body::MoveKinematic` or `BodyInterface::MoveKinematic`.
